praw/models/reddit/subreddit.py

```diff
@@ -1545,6 +1545,41 @@ def __init__(self, subreddit):
         """Construct an instance of the Modmail object."""
         self.subreddit = subreddit
 
+    def _build_subreddit_list(self, other_subreddits):
+        """Return a comma-separated list of subreddit display names."""
+        subreddits = [self.subreddit] + (other_subreddits or [])
+        return ",".join(str(subreddit) for subreddit in subreddits)
+
+    def bulk_read(self, other_subreddits=None, state=None):
+        """Mark conversations for subreddit(s) as read.
+
+        Due to server-side restrictions, 'all' is not a valid subreddit for
+        this method. Instead, use :meth:`~.Modmail.subreddits` to get a list of
+        subreddits using the new modmail.
+
+        :param other_subreddits: A list of `.Subreddit` instances for which to
+            mark conversations (default: None).
+        :param state: Can be one of: all, archived, highlighted, inprogress,
+            mod, new, notifications, (default: all).
+        :returns: A generator of :class:`.ModmailConversation` instances that
+            were marked read.
+
+        For example, to mark all notifications for a subreddit as read:
+
+        .. code:: python
+
+           subreddit = reddit.subreddit('redditdev')
+           subreddit.modmail.bulk_read(state='notifications')
+
+        """
+        params = {'entity': self._build_subreddit_list(other_subreddits)}
+        if state:
+            params['state'] = state
+        response = self.subreddit._reddit.post(
+            API_PATH['modmail_bulk_read'], params=params)
+        return (self(conversation_id)
+                for conversation_id in response['conversation_ids'])
```

Comment:
I made this a generator just in case, but I don't know how plausible it is that this response would ever be paginated. After all, the whole point of this endpoint is to archive a whole set of messages with a single API call.